---
layout: article
titles:
  en      : &EN       Microbiome amplicon analyses
  en-GB   : *EN
  en-US   : *EN
  en-CA   : *EN
  en-AU   : *EN
key: page-Microbiome
---

* This tutorial outlines the basics concepts in Plasmodium amplicon (i.e. 16S) analyses
	
## Learning outcomes

* Explain various terminology and concepts associated with microbiomes
* Recognise the three primary diversity measures
* Compare and critique various methodologies for microbiome profiling
* Execute data cleaning and diversity measure analyses using QIIME-2


## Prerequisites

* It is recommended that you have [Notepad++ (Windows)](https://notepad-plus-plus.org/downloads/) or [BBEdit (Mac)](https://www.barebones.com/products/bbedit/) for viewing fasta files; most linux default editors can do this.
* It is recommended that you have followed the [Concepts in Computer Programming](https://conmeehan.github.io/PathogenDataCourse/ConceptsInComputerProgramming), [UNIX tutorial (basics)](https://conmeehan.github.io/UNIXtutorial) and [Setting up and using conda](https://conmeehan.github.io/PathogenDataCourse/Worksheets/CondaInstallAndUse) tutorials if you are going to do the QIIME-2 worksheet.

## Approximate time to finish tutorial
* Lecture: 30 mins
* Tutorials: 1 hour
* Pre/post surveys: 10 minutes

## Order of tutorial

Please do the pre-learning quiz, then watch the presentation. <br />
Once finished the tutorial, take the post-learing quiz.<br />


### <a href="https://ntusurvey.onlinesurveys.ac.uk/microbiome-analysis-pre-tutorial-survey" target="_blank">Microbiome analysis Pre-tutorial Survey</a>


## Presentation

* [Download slides here](https://conmeehan.github.io/PathogenDataCourse/SlideSets/MicrobiomeAnalysis.pptx)


## Worksheets
### [Microbiome 16S amplicon sequencing processing with QIIME-2 (via UNIX)](https://conmeehan.github.io/PathogenDataCourse/Worksheets/Microbiome_QIIME2)
### [Microbiome 16S amplicon sequencing processing with Mothur (via Galaxy)](https://training.galaxyproject.org/training-material/topics/metagenomics/)


### <a href="https://ntusurvey.onlinesurveys.ac.uk/microbiome-analysis-post-tutorial-survey" target="_blank">Microbiome analysis Post-tutorial Survey</a>


## Other tutorials/tools
* [Mothur tutorial](https://mothur.org/wiki/miseq_sop/)
	* An alternative to QIIME-2 for 16S amplicon sequenced microbiome analyses
* Various QIIME-2 tutolorials:
	* [“Atacama soil microbiome” tutorial](https://docs.qiime2.org/2023.5/tutorials/atacama-soils/)
	* [John Quensen tutorial](https://john-quensen.com/tutorials/processing-16s-sequences-with-qiime2-and-dada2/)
	* [Youtube video tutorial](https://www.youtube.com/watch?v=iqz1c2QhUD8&list=PLbVDKwGpb3XmvnTrU40zHRT7NZWWVNUpt&index=1)
* [Open & Reproducible Microbiome Data Analysis Spring School 2018](https://mibwurrepo.github.io/Microbial-bioinformatics-introductory-course-Material-2018/introduction.html)	