---
layout: article
titles:
  en      : &EN       Pathogen genomics data analysis course
  en-GB   : *EN
  en-US   : *EN
  en-CA   : *EN
  en-AU   : *EN
key: page-PathogenCourse
---


This workshop contains a set of tutorials for undertaking basic data analytics for pathogen genomics research (although many topics covered are also applicable elsewhere).<br />
It is made to be lightweight, with each tutorial capable of running on a laptop. The primary target audience is public health researchers/technicians in low resource settings but there is material here for anybody starting out in bioinformatics and/or pathogen genomics data analysis  <br/>

<p xmlns:cc="http://creativecommons.org/ns#" >This work is licensed under <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/?ref=chooser-v1" target="_blank" rel="license noopener noreferrer" style="display:inline-block;">CC BY-NC-SA 4.0<img style="height:22px!important;margin-left:3px;vertical-align:text-bottom;" src="https://mirrors.creativecommons.org/presskit/icons/cc.svg?ref=chooser-v1"><img style="height:22px!important;margin-left:3px;vertical-align:text-bottom;" src="https://mirrors.creativecommons.org/presskit/icons/by.svg?ref=chooser-v1"><img style="height:22px!important;margin-left:3px;vertical-align:text-bottom;" src="https://mirrors.creativecommons.org/presskit/icons/nc.svg?ref=chooser-v1"><img style="height:22px!important;margin-left:3px;vertical-align:text-bottom;" src="https://mirrors.creativecommons.org/presskit/icons/sa.svg?ref=chooser-v1"></a></p>
That means it is fine to reuse, distribute, adapt and build upon the material for non-commercial purposes, once proper credit is given to the relevant creator (i.e. Conor Meehan for most material). If you adapt or modify the material, you must license the modified material under the same terms. <br />
If you do reuse or adapt the material, please let me know as I am always eager to hear if the material is useful. My contact details can be found on my [Nottingham trent University staff page](https://www.ntu.ac.uk/staff-profiles/science-technology/conor-meehan). <br />
<br/>
Funding for the basic creation of this course comes from the [Microbiology Society](https://microbiologysociety.org/)<br/>


Each tutorial has a lecture or explanation of the topic with associated practice worksheets and sample answers.<br />
It is suggested that you do these in order, or at least the concepts in computer programming, UNIX tutorials and R tutorial before the other topics, if you wish to do the terminal and/or R-based worksheets (there are Galaxy and web-based alternatives for many if you do not). <br />
## Topics (each is a hyperlink to its own page):
### [Concepts in computer programming](https://conmeehan.github.io/PathogenDataCourse/ConceptsInComputerProgramming)
### [The UNIX shell](https://conmeehan.github.io/PathogenDataCourse/UNIXshell)
### [Introduction to R](https://conmeehan.github.io/PathogenDataCourse/IntroToR)
### [Experimental design and statistics](https://conmeehan.github.io/PathogenDataCourse/Statistics)
### [Data handling and figure generation](https://conmeehan.github.io/PathogenDataCourse/DataHandlingAndPresentation)
### [Biological Databases and BLAST](https://conmeehan.github.io/PathogenDataCourse/BiologicalDBsAndBLAST)
### [Genome assembly and annotation (viral and bacterial)](https://conmeehan.github.io/PathogenDataCourse/GenomeAssemblyAnnotation)
### [Introduction to phylogenetics](https://conmeehan.github.io/PathogenDataCourse/IntroToPhylogenetics)
### [Introduction to Bayesian phylodynamics](https://conmeehan.github.io/PathogenDataCourse/IntroToPhylodynamics)
### [Predicting pathogenic features (AMR, virulence factors, plasmids)](https://conmeehan.github.io/PathogenDataCourse/PredictingPathogenicFeatures)
### [Plasmodium genomics and drug resistance](https://conmeehan.github.io/PathogenDataCourse/PlasmodiumGenomics)
### [Bacterial genomic epidemiology and strain typing](https://conmeehan.github.io/PathogenDataCourse/GenomicEpiTyping)
### [Microbiome amplicon analyses](https://conmeehan.github.io/PathogenDataCourse/MicrobiomeAnalysis)
<br />

## Additional (related) useful tutorials and resources (by other people)
* [Galaxy tutorials](https://training.galaxyproject.org/)
   * Various tutorials from introduction to advanced analyses using the Galaxy portal.
   * Many of the Galaxy tutorials in this workshop comes from this set of tutorials.
* [Swiss Bioinformatics institute training material](https://github.com/sib-swiss/training-collection)
   * Extensive computer programing training for bioinformatics
* [Bactopia workshop](https://github.com/lskatz/bactopia-workshop/tree/main/workshop) <br />
    * Guidance on running the complete bacterial genome assembly and analysis pipeline [Bactopia](https://bactopia.github.io/v2.2.0/)
* [Proposed Standards for Public Health Bioinformatics Software (PHA4GE)](https://github.com/pha4ge/pipeline-resources/blob/main/docs/pipeline-standards.md)
	* A comprehensive set of guidelines for creating software for public health use
	
### Python tutorials
* [W3 schools python tutorial](https://www.w3schools.com/python/default.asp)
* [UC Davis bioinformatics core introduction to Python](https://ucdavis-bioinformatics-training.github.io/2022-Feb-Introduction-To-Python-For-Bioinformatics/python/python1)
* [Python for biologists](https://www.pythonforbiologists.org/)
* [Full spectrum bioinformatics](https://github.com/zaneveld/full_spectrum_bioinformatics)
   * Introduction to bioinformatics via Python coding
    


