---
layout: article
titles:
  en      : &EN       Predicting pathogenic features
  en-GB   : *EN
  en-US   : *EN
  en-CA   : *EN
  en-AU   : *EN
key: page-PathogenicFeatures
---


This tutorial outlines the various online biological databases and their uses as well as an introduction to homology terminology and searching using BLAST.<br />

## Learning outcomes
* 


## Prerequisites
* It is recommended that you have [Notepad++ (Windows)](https://notepad-plus-plus.org/downloads/) or [BBEdit (Mac)](https://www.barebones.com/products/bbedit/) for viewing fasta files; most linux default editors can do this.
* It is recommended that you have followed the [Concepts in Computer Programming](https://conmeehan.github.io/PathogenDataCourse/ConceptsInComputerProgramming), [UNIX tutorial (basics)](https://conmeehan.github.io/UNIXtutorial) and [Setting up and using conda](https://conmeehan.github.io/PathogenDataCourse/Worksheets/CondaInstallAndUse) tutorials if you are going to do the UNIX-based worksheets.

## Approximate time to finish tutorial
* Lecture: 20 mins
* Tutorials: 30 mins
* Pre/post surveys: 10 minutes


## Order of tutorial

Please do the pre-learning quiz, then watch the presentation. <br />
During the presentation there are points to stop and do exercises, which are linked below. The answers to the questions in the exercises are linked within each one.<br />
Once finished the tutorial, take the post-learing quiz.<br />

### <a href="https://ntusurvey.onlinesurveys.ac.uk/predictng-pathogenic-features-pre-tutorial-survey" target="_blank">Genome Assembly Pre-tutorial Survey</a>


## Presentation
* [Download slides here](https://conmeehan.github.io/PathogenDataCourse/SlideSets/PredictingPathogenicFeatures.pptx)

## Worksheets
### [Predicting pathogenic features using AbritAMR and PlasmidFinder via UNIX](https://conmeehan.github.io/PathogenDataCourse/Worksheets/Predicting_AMR_VF_Plasmids_UNIX)
### [Predicting pathogenic features using AbritAMR and PlasmidFinder via Galaxy](https://conmeehan.github.io/PathogenDataCourse/Worksheets/Predicting_AMR_VF_Plasmids_Galaxy)


### <a href="https://ntusurvey.onlinesurveys.ac.uk/predictng-pathogenic-features-post-tutorial-survey" target="_blank">Genome Assembly Post-tutorial Survey</a>


### Other tools and videos
* [PlasForest](https://github.com/leaemiliepradier/PlasForest)
  * Random forest based classification of plasmids
* Antismash
  * Tool for finding secondary metabolites in genome assemblies 
* [IslandViewer 4](https://www.pathogenomics.sfu.ca/islandviewer/upload/)
	* Tool for finding genomic (including pathogenicity) islands  
